src/overlays/actors/ovl_Bg_Jya_Bombchuiwa/z_bg_jya_bombchuiwa.c

```diff
@@ -21,31 +27,211 @@ const ActorInit Bg_Jya_Bombchuiwa_InitVars = {
+    }
+}
+
+void BgJyaBombchuiwa_CleanUpAfterExplosion(BgJyaBombchuiwa* this, GlobalContext* globalCtx) {
+    this->actionFunc = func_808949B8;
+    BgJyaBombchuiwa_SetDrawFlag(this, 4);
+    this->lightRayIntensity = 0.3f;
+    this->timer = 0;
+    this->actor.flags &= ~1;
+}
+
+void func_808949B8(BgJyaBombchuiwa* this, GlobalContext* globalCtx) {
+    this->timer++;
+    if (this->timer & 4) {
+        func_80033480(globalCtx, &this->actor.posRot.pos, 60.0f, 3, 100, 100, 0);
+    }
+    if (Math_ApproxF(&this->lightRayIntensity, 1.0f, 0.028)) {
+        BgJyaBombchuiwa_SpawnLightRay(this, globalCtx);
+    }
+}
+
+void BgJyaBombchuiwa_SpawnLightRay(BgJyaBombchuiwa* this, GlobalContext* globalCtx) {
+    this->actionFunc = NULL;
+    this->lightRayIntensity = 153.0f;
+    BgJyaBombchuiwa_SetDrawFlag(this, 4);
+    if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_MIR_RAY, this->actor.posRot.pos.x, this->actor.posRot.pos.y,
+                    this->actor.posRot.pos.z, 0, 0, 0, 0) == NULL) {
+        // Occurrence failure
+        osSyncPrintf("Ｅｒｒｏｒ : Mir_Ray 発生失敗(%s %d)(arg_data 0x%04x)\n", "../z_bg_jya_bombchuiwa.c", 410,
+                     this->actor.params);
+    }
+}
+
+void BgJyaBombchuiwa_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaBombchuiwa* this = THIS;
+
+    if (this->actionFunc != NULL) {
+        this->actionFunc(this, globalCtx);
+    }
+}
+
+void BgJyaBombchuiwa_DrawRock(GlobalContext* globalCtx) {
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_jya_bombchuiwa.c", 436);
+    func_80093D84(globalCtx->state.gfxCtx);
+    gSPMatrix(oGfxCtx->polyXlu.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_jya_bombchuiwa.c", 0x1B7),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(oGfxCtx->polyXlu.p++, D_060119B0);
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_bg_jya_bombchuiwa.c", 443);
+}
+
+void BgJyaBombchuiwa_DrawLight(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaBombchuiwa* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_jya_bombchuiwa.c", 453);
+    func_80093D84(globalCtx->state.gfxCtx);
+    gSPMatrix(oGfxCtx->polyXlu.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_jya_bombchuiwa.c", 457),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gDPSetPrimColor(oGfxCtx->polyXlu.p++, 0, 0, 255, 255, 255, CLAMP_MAX((u32)(this->lightRayIntensity * 153.0f), 153));
+    gSPDisplayList(oGfxCtx->polyXlu.p++, D_0600DC40);
+    gDPPipeSync(oGfxCtx->polyXlu.p++);
+    gDPSetPrimColor(oGfxCtx->polyXlu.p++, 0, 0, 255, 255, 255, CLAMP_MAX((u32)(this->lightRayIntensity * 255.0f), 255));
+    gSPDisplayList(oGfxCtx->polyXlu.p++, D_0600DB60);
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_bg_jya_bombchuiwa.c", 472);
+}
+
+/* Uses drawFlags to determine how to draw the object.  If the first 2 bits are set it draws the rock.  If the 3rd bit
+ * is set it will not draw the rock but will instead draw a beam of light
+ */
```

Comment:
This isn't necessary as a comment imo. Also it doesn't seem to really correspond to what the function does atm, but even if it did it would just be redundant imo since the code is already very easy to read here.

However, if you really want to document what the flags do, I suggest making defines with proper names for them.